src/Init/Data/List/Lemmas.lean

```diff
@@ -1421,7 +1574,7 @@ theorem take_left : ∀ l₁ l₂ : List α, take (length l₁) (l₁ ++ l₂) =
 theorem take_left' {l₁ l₂ : List α} {n} (h : length l₁ = n) : take n (l₁ ++ l₂) = l₁ := by
   rw [← h]; apply take_left
 
-theorem map_take (f : α → β) :
+@[simp] theorem map_take (f : α → β) :
```

Comment:
Is map simpler or take? Hard to say; the RHS applies f to elements that are thrown away. Maybe it doesn't matter much, but maybe the section describing the simp setup could have a list of all functions ordered by how the simp set will sort them?

Reply:
Here is a possible rationale: `map` has a higher chance to simplify with list producer functions, so it's worth moving it in.

In general maybe it's more likely for things to happen further in, so functions with the ”better” simp set, i.e. more rules, like simp, should be moved inside.

Reply:
I think this is another example of "prefer moving functions inwards past 'structural' transformations", on the principle that the function will eventually hit the elements of the list and compute. I've updated the docs to reflect this.